[2025-03-24T09:04:20.736+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: fetch_filtered_data_and_upload_to_gcs.fetch_and_upload_csv_files manual__2025-03-24T09:04:18.274655+00:00 [queued]>
[2025-03-24T09:04:20.747+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: fetch_filtered_data_and_upload_to_gcs.fetch_and_upload_csv_files manual__2025-03-24T09:04:18.274655+00:00 [queued]>
[2025-03-24T09:04:20.747+0000] {taskinstance.py:2193} INFO - Starting attempt 1 of 1
[2025-03-24T09:04:20.760+0000] {taskinstance.py:2214} INFO - Executing <Task(PythonOperator): fetch_and_upload_csv_files> on 2025-03-24 09:04:18.274655+00:00
[2025-03-24T09:04:20.763+0000] {standard_task_runner.py:60} INFO - Started process 1142 to run task
[2025-03-24T09:04:20.767+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'fetch_filtered_data_and_upload_to_gcs', 'fetch_and_upload_csv_files', 'manual__2025-03-24T09:04:18.274655+00:00', '--job-id', '110', '--raw', '--subdir', 'DAGS_FOLDER/step_one.py', '--cfg-path', '/tmp/tmpgzj45d33']
[2025-03-24T09:04:20.770+0000] {standard_task_runner.py:88} INFO - Job 110: Subtask fetch_and_upload_csv_files
[2025-03-24T09:04:20.815+0000] {task_command.py:423} INFO - Running <TaskInstance: fetch_filtered_data_and_upload_to_gcs.fetch_and_upload_csv_files manual__2025-03-24T09:04:18.274655+00:00 [running]> on host 3d764bfd33f3
[2025-03-24T09:04:20.895+0000] {taskinstance.py:2510} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='fetch_filtered_data_and_upload_to_gcs' AIRFLOW_CTX_TASK_ID='fetch_and_upload_csv_files' AIRFLOW_CTX_EXECUTION_DATE='2025-03-24T09:04:18.274655+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-03-24T09:04:18.274655+00:00'
[2025-03-24T09:04:20.897+0000] {logging_mixin.py:188} INFO - Fetching data from: https://cycling.data.tfl.gov.uk/ActiveTravelCountsProgramme/
[2025-03-24T09:04:21.943+0000] {taskinstance.py:2728} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 444, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 414, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 200, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 217, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/step_one.py", line 68, in fetch_and_upload_csv_files
    csv_urls = get_csv_urls_from_website(base_url, subdirectory_url)
  File "/opt/airflow/dags/step_one.py", line 30, in get_csv_urls_from_website
    raise Exception(f"Failed to fetch the website. Status code: {response.status_code}")
Exception: Failed to fetch the website. Status code: 404
[2025-03-24T09:04:21.952+0000] {taskinstance.py:1149} INFO - Marking task as FAILED. dag_id=fetch_filtered_data_and_upload_to_gcs, task_id=fetch_and_upload_csv_files, execution_date=20250324T090418, start_date=20250324T090420, end_date=20250324T090421
[2025-03-24T09:04:21.965+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 110 for task fetch_and_upload_csv_files (Failed to fetch the website. Status code: 404; 1142)
[2025-03-24T09:04:21.985+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-03-24T09:04:22.000+0000] {taskinstance.py:3309} INFO - 0 downstream tasks scheduled from follow-on schedule check
